addons/web_editor/models/ir_qweb.py

```diff
@@ -47,10 +47,25 @@ class QWeb(models.AbstractModel):
 
     # compile directives
 
+    def _compile_node(self, el, options):
+        if options.get('is-snippet') == options['template']:
+            # Get the path of the template's content
+            path = options['root'].getpath(el).replace('/t', '')
+            # Only append the template key on template's first node.
+            # If it already has a data-snippet it is a saved snippet.
+            # Do not override it.
+            size = len(path.split('/'))
+            if size == 2 and 'data-snippet' not in el.attrib:
+                el.attrib['data-snippet'] = options['template'].split('.')[-1]
```

Comment:
Feels like it should be more robust, someone can call its snippet `my_app.hello.world.test.hey`

Reply:
Yes, `el.attrib['data-snippet'] = options['template'].split('.', 1)[-1]` looks better than